Build break on latest develop branch when ENABLE_SECURITY is OFF

### Bug Report
Build break on latest develop branch when ENABLE_SECURITY is OFF

##### Reproduction Steps
1. Try to build SDL with cmake ` -DENABLE_SECURITY=OFF` flag

##### Expected Behavior
SDL is built successfully

##### Observed Behavior
Build failed

##### OS & Version Information
* OS/Version: Ubuntu 16.04 amd64
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop (e90b4aa)

##### Test Case, Sample Code, and / or Example App
- Console log
```
[ 13%] Building CXX object src/components/connection_handler/CMakeFiles/connectionHandler.dir/src/connection.cc.o
/home/db/ramdrv/sdl_core/src/components/protocol_handler/src/protocol_handler_impl.cc: In member function ‘virtual void protocol_handler::ProtocolHandlerImpl::NotifyOnGetSystemTimeFailed()’:
/home/db/ramdrv/sdl_core/src/components/protocol_handler/src/protocol_handler_impl.cc:1134:3: error: ‘security_manager_’ was not declared in this scope
   security_manager_->ResetPendingSystemTimeRequests();
```

--- a/src/components/application_manager/include/application_manager/application_manager_impl.h
+++ b/src/components/application_manager/include/application_manager/application_manager_impl.h
@@ -584,7 +584,9 @@ class ApplicationManagerImpl
       utils::Optional<hmi_apis::Common_ServiceStatusUpdateReason::eType>
           service_update_reason) FINAL;
 
+#ifdef ENABLE_SECURITY
   bool OnPTUFailed() FINAL;
+#endif  // ENABLE_SECURITY
   /*
    * @brief Starts audio pass thru thread
    *

--- a/src/components/application_manager/src/application_manager_impl.cc
+++ b/src/components/application_manager/src/application_manager_impl.cc
@@ -2163,11 +2163,12 @@ bool ApplicationManagerImpl::OnHandshakeDone(
   }
   return false;
 }
-
+#ifdef ENABLE_SECURITY
 bool ApplicationManagerImpl::OnPTUFailed() {
   LOG4CXX_AUTO_TRACE(logger_);
   return false;
 }
+#endif  // ENABLE_SECURITY
 
 bool ApplicationManagerImpl::OnGetSystemTimeFailed() {
   LOG4CXX_AUTO_TRACE(logger_);

--- a/src/components/application_manager/src/rpc_service_impl.cc
+++ b/src/components/application_manager/src/rpc_service_impl.cc
@@ -164,6 +164,7 @@ bool RPCServiceImpl::ManageMobileCommand(
       SendMessageToMobile(response);
       return false;
     }
+#ifdef ENABLE_SECURITY
     if (EncryptionFlagCheckResult::kError_EncryptionNeeded ==
         IsEncryptionRequired(
             *message,
@@ -177,6 +178,7 @@ bool RPCServiceImpl::ManageMobileCommand(
       SendMessageToMobile(response);
       return false;
     }
+#endif  // ENABLE_SECURITY
 
     // Message for "CheckPermission" must be with attached schema
     mobile_so_factory().attachSchema(*message, false);

--- a/src/components/protocol_handler/src/protocol_handler_impl.cc
+++ b/src/components/protocol_handler/src/protocol_handler_impl.cc
@@ -1131,14 +1131,16 @@ void ProtocolHandlerImpl::OnUnexpectedDisconnect(
 
 void ProtocolHandlerImpl::NotifyOnGetSystemTimeFailed() {
   LOG4CXX_AUTO_TRACE(logger_);
-  security_manager_->ResetPendingSystemTimeRequests();
 #ifdef ENABLE_SECURITY
+  security_manager_->ResetPendingSystemTimeRequests();
   security_manager_->NotifyListenersOnGetSystemTimeFailed();
 #endif  // ENABLE_SECURITY
 }
 
 void ProtocolHandlerImpl::ProcessFailedPTU() {
+#ifdef ENABLE_SECURITY
   security_manager_->ProcessFailedPTU();
+#endif  // ENABLE_SECURITY
 }
 
 #ifdef EXTERNAL_PROPRIETARY_MODE
@@ -1228,10 +1230,14 @@ void ProtocolHandlerImpl::OnAuthTokenUpdated(const std::string& policy_app_id,
 bool ProtocolHandlerImpl::IsRPCServiceSecure(
     const uint32_t connection_key) const {
   LOG4CXX_AUTO_TRACE(logger_);
+#ifdef ENABLE_SECURITY
 
   security_manager::SSLContext* context =
       session_observer_.GetSSLContext(connection_key, ServiceType::kRpc);
   return (context && context->IsInitCompleted());
+#else
+  return false;
+#endif  // ENABLE_SECURITY
 }
 
 RESULT_CODE ProtocolHandlerImpl::SendFrame(const ProtocolFramePtr packet) {
@@ -1617,6 +1623,7 @@ const ServiceStatus ProtocolHandlerImpl::ServiceDisallowedBySettings(
       std::find(audio_transports.begin(), audio_transports.end(), transport) !=
           audio_transports.end();
 
+#ifdef ENABLE_SECURITY
   const auto& force_protected = get_settings().force_protected_service();
 
   const auto& force_unprotected = get_settings().force_unprotected_service();
@@ -1631,17 +1638,18 @@ const ServiceStatus ProtocolHandlerImpl::ServiceDisallowedBySettings(
 
   const bool can_start_unprotected = is_force_unprotected && !protection;
 
-  if ((ServiceType::kMobileNav == service_type && !is_video_allowed) ||
-      (ServiceType::kAudio == service_type && !is_audio_allowed)) {
-    return ServiceStatus::SERVICE_START_FAILED;
-  }
-
   if (is_force_protected && !can_start_protected) {
     return ServiceStatus::PROTECTION_ENFORCED;
   }
 
   if (is_force_unprotected && !can_start_unprotected) {
     return ServiceStatus::UNSECURE_START_FAILED;
+  }
+#endif  // ENABLE_SECURITY
+
+  if ((ServiceType::kMobileNav == service_type && !is_video_allowed) ||
+      (ServiceType::kAudio == service_type && !is_audio_allowed)) {
+    return ServiceStatus::SERVICE_START_FAILED;
   }
 
   return ServiceStatus::INVALID_ENUM;

--- a/src/components/protocol_handler/test/protocol_handler_tm_test.cc
+++ b/src/components/protocol_handler/test/protocol_handler_tm_test.cc
@@ -746,12 +746,11 @@ TEST_F(ProtocolHandlerImplTest,
   const uint8_t session_id1 = 1u;
   const ::transport_manager::ConnectionUID connection_id2 = 0xBu;
   const uint8_t session_id2 = 2u;
+  EXPECT_CALL(session_observer_mock, KeyFromPair(connection_id2, session_id2))
+      .WillRepeatedly(Return(connection_key));
 
 #ifdef ENABLE_SECURITY
   AddSecurityManager();
-
-  EXPECT_CALL(session_observer_mock, KeyFromPair(connection_id2, session_id2))
-      .WillRepeatedly(Return(connection_key));
 
   EXPECT_CALL(session_observer_mock,
               GetSSLContext(connection_key, start_service))

--- a/src/components/transport_manager/test/websocket_connection_test.cc
+++ b/src/components/transport_manager/test/websocket_connection_test.cc
@@ -425,6 +425,7 @@ TEST_F(WebsocketConnectionTest, WSSConnection_SUCCESS_ValidTarget) {
   t1.join();
 }
 
+#ifdef ENABLE_SECURITY
 TEST_F(WebsocketConnectionTest, WSSConnection_FAILURE_InvalidTarget) {
   transport_manager::transport_adapter::CloudAppProperties properties{
       .endpoint = "wss://" + kHost + ":" + std::to_string(kPort),
@@ -486,6 +487,7 @@ TEST_F(WebsocketConnectionTest, WSSConnection_FAILURE_IncorrectCert) {
   wss_session->Stop();
   t1.join();
 }
+#endif  // ENABLE_SECURITY
 }  // namespace transport_manager_test
 }  // namespace components
 }  // namespace test

